### DISCUSSION TITLE
RollupDateLiteralTest Failing at Fiscal Quarter

### DISCUSSION THREAD

**Discussion by shane-saltbox** [jamessimone/apex-rollup#324] [Q&A]
@jamessimone I'm attaching a quick analysis of what we found while trying to manually implement the test coverage for the package. We'd like to be able to use the plug-in now that I know it's there, but suspect that'll fail until we're able to address the issue we found. 

For context, our business starts it's fiscal year in February, which has also been updated on the 'Fiscal Year Starts In' value to February. 

While going through the install of the test classes several methods in RollupDateLiteralTest failed due to the use of THIS_QUARTER instead of THIS_FISCAL_QUARTER for example. We were able to get around most of them quickly by utilizing the fiscal date, but also had to make a few other changes / comment a few sections out. At the end of the day there was still a good code coverage after these changes. 

I'm attaching a quick diff so it's easy for you to find what changes we had to make. 

![screencapture-github-shane-saltbox-saltbox-sfdc-pull-43-files-2022-05-24-18_34_22](https://user-images.githubusercontent.com/89033396/170149588-62d921e0-c8cf-4ce5-bbb3-e41a23163ac8.png)

Let me know if you had any questions!

**Comment by ssk42** [comment]
I was able to successfully reproduce this issue by changing the Fiscal Year Start Month to February where shouldWorkForFiscalMonthFunction, shouldWorkForFiscalQuarterFunction, shouldWorkForLastFiscalYear, shouldWorkForLastNFiscalYears, shouldWorkForLastNQuarters, shouldWorkForLastQuarter, shouldWorkForNFiscalYearsAgo, shouldWorkForNQuartersAgo, shouldWorkForNextFiscalYear, shouldWorkForNextNFiscalYears, shouldWorkForNextNQuarters, shouldWorkForNextQuarter, shouldWorkForThisFiscalYear, shouldWorkForThisQuarter all failed.  Were there any other additional changes or fails that I missed?

**Reply by jamessimone** [reply]
I made #325 to track this one

**Reply by winfotech** [reply]
@ssk42 You covered all of them, Also attaching the Test Class with the temporary fixes for your reference.
[RollupDateLiteralTests.cls.txt](https://github.com/jamessimone/apex-rollup/files/8768841/RollupDateLiteralTests.cls.txt)
